src/lib/autoinstall/autosetup_helpers.rb

```diff
@@ -126,23 +126,13 @@ def autosetup_network
       # Prevent to be called twice in case of already configured
       return if @network_configured
 
-      if Yast::Profile.current["networking"]
-        if network_before_proposal?
-          log.info("Networking setup before the proposal")
-        else
-          log.info("Networking setup at the end of first installation stage")
-        end
-
-        log.info("Importing Network settings from configuration file")
-        Yast::WFM.CallFunction("lan_auto", ["Import", Yast::Profile.current["networking"]])
-        Yast::Profile.remove_sections("networking")
-
-        # Import also the host section in order to resolve hosts only available
-        # with the network configuration and the host entry
-        if Yast::Profile.current["host"] && network_before_proposal?
-          Yast::WFM.CallFunction("host_auto", ["Import", Yast::Profile.current["host"]])
-          Yast::Profile.remove_sections("host")
-        end
+      networking_section = Yast::Profile.current.fetch("networking", {})
+      Yast::WFM.CallFunction("lan_auto", ["Import", networking_section])
+
+      # Import also the host section in order to resolve hosts only available
+      # with the network configuration and the host entry
+      if Yast::Profile.current["host"] && network_before_proposal?
```

Comment:
```suggestion
      if Yast::Profile.current["host"]
```

Reply:
As the remove of the sections are done unconditionally, then we should do the host import always. I already planned to do this change, so, it should be ok.